### PR TITLE
[DOCKER] ADD PG Auto Failover

### LINKED ISSUE #8: PG Auto Failover
[This](https://pg-auto-failover.readthedocs.io/en/main/) plugin needs to be integrated to the project.

### PR DESCRIPTION
**Pull request by PiyushRaj927**
Fixes #8 

1. Installed the required build dependency.
2. Downloaded pgautofailover v2.0 from https://github.com/hapostgres/pg_auto_failoverand built it.
3. Installed pgautofailover.
4. Cleaned up the build dependencies.
5. Updated the smoke test for pgautofailover. 

## Testing
I have test the images locally and have found no issues

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -281,4 +281,37 @@ RUN set -eux \
...
+        libxslt-dev \
+        llvm15 \
+# build pg_auto_failover
+    && wget  -O /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION}.zip "https://github.com/hapostgres/pg_auto_failover/archive/refs/tags/v${PG_AUTO_FAILOVER_VERSION}.zip" \
+    && unzip  /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION}.zip -d /tmp \
+    && ls -alh /tmp \
+    && cd /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION} \
+    && make \
+    && make install \
+    && sed -i "s/\(shared_preload_libraries = '[^']*\)/\1,pgautofailover/"  /usr/local/share/postgresql/postgresql.conf.sample \
```
Comment (jayanth-kumar-morem) [inline Dockerfile:312]: We are already adding citus,postgis,pg_vector etc to shared_preload_libraries, can you add pgautofailover over there itself

File: bitnami/Makefile [hunk 2]
```diff
@@ -6,8 +6,9 @@ PG_VER=pg15
 CITUS_VERSION="11.2.0"
 POSTGIS_MAJOR=3
 PG_MAJOR=15
-POSTGIS_VERSION=3.3.2+dfsg-1.pgdg110+1
+POSTGIS_VERSION=3.3.3+dfsg-1~exp1.pgdg110+1
```
Comment (jayanth-kumar-morem) [inline bitnami/Makefile:9]: What does `pgdg110` refer to ?

File: Dockerfile (hunk 1 above)
Comment (jayanth-kumar-morem) [inline Dockerfile:312]: We are already adding citus,postgis,pg_vector etc to shared_preload_libraries, can you add pgautofailover over there itself

**Comment by jayanth-kumar-morem** [comment]
Hi @PiyushRaj927 , The code looks clean, gave a few minor comments.

**Comment by PiyushRaj927** [comment]
Hey, @jayanth-kumar-morem, I'm having trouble viewing your comments. Could you please double-check them?

**Review by jayanth-kumar-morem — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (PiyushRaj927) [reply Dockerfile:312]: Sure

File: bitnami/Makefile (hunk 2 above)
Comment (PiyushRaj927) [reply bitnami/Makefile:9]: `pgdg110` is [ PostgreSQL Global Development Group apt repository](https://www.postgresql.org/about/news/pgdg-apt-repository-for-debianubuntu-1432/) for Debian 11

Comment (jayanth-kumar-morem) [reply bitnami/Makefile:9]: `bitnami/postgresql` images uses `minideb`. Minideb is a minimalist Debian-based image built specifically to be used as a base image for containers(https://github.com/bitnami/minideb). Please confirm compatibility.

Comment (PiyushRaj927) [reply bitnami/Makefile:9]: The smoke test is passing https://github.com/Samagra-Development/WarpSQL/actions/runs/5142018741/jobs/9255221212?pr=40#step:5:103
Can you provide more details or specify any additional tests you would like me to perform?

**Review by jayanth-kumar-morem — APPROVED** [review]
Looks Good To Me

**Pull request merged**